src/Rule/Ip.php

```diff
@@ -275,10 +306,15 @@ public function validateValue($value): Result
      * array  - an error occurred during the validation.
      * Array[0] contains the text of an error, array[1] contains values for the placeholders in the error message
      */
-    private function validateSubnet(string $ip)
+    private function validateSubnet(string &$ipAddress)
```

Comment:
What is the purpose of & here?

Reply:
Initially this method returned `array|string`. `Array` in case of error, and `string` with formatted value, in case of success.
Now we return `Result()`. `&` is used to set the formatted value directly to the parameter.

Reply:
What shall I do then? Remove unnecessary tests and functionality?

Reply:
Yes. See https://forum.yiiframework.com/t/saving-or-killing-non-validation-in-validators/126086